src/raster/r.viewshed.exposure/r.viewshed.exposure.py

```diff
@@ -0,0 +1,1409 @@
+        clean_temp(vs.pid)
+
+        np_viewshed = raster2numpy(tmp_vs).astype(np.single)
+        np_viewshed[np_viewshed == -2147483648] = np.nan
+
+        # ======================================================================
+        # 3. Prepare local coordinates and attributes of target point T
+        # ======================================================================
+        # Calculate how much of rows/cols of local region lies
+        # outside global region
+        o_1 = [
+            max(t_glob[1] + exp_range + reg.nsres / 2 - reg.north, 0),
+            max(reg.west - (t_glob[0] - exp_range - reg.ewres / 2), 0),
+        ]
+
+        t_loc = np.append(
+            np.array(
+                [
+                    exp_range / reg.nsres + 0.5 - o_1[0] / reg.nsres,
+                    exp_range / reg.ewres + 0.5 - o_1[1] / reg.ewres,
+                ]
+            ),
+            t_glob[2:],
+        )
+
+        # ======================================================================
+        # 4. Parametrise viewshed
+        # ======================================================================
+        np_viewshed = parametrise_viewshed(
+            lreg_shape,
+            t_loc,
+            np_viewshed,
+            reg,
+            exp_range,
+            r_dsm,
+            dsm_type,
+            v_elevation,
+            b_1,
+        ).astype(np.single)
+
+        # ======================================================================
+        # 5. Cummulate viewsheds
+        # ======================================================================
+        # Determine position of local parametrised viewshed within
+        # global cummulative viewshed
+        o_2 = [
+            int(round((reg.north - loc_reg_n) / reg.nsres)),  # NS (rows)
+            int(round((loc_reg_w - reg.west) / reg.ewres)),  # EW (cols)
+        ]
+
+        # Add local parametrised viewshed to global cummulative viewshed
+        # replace nans with 0 in processed regions, keep nan where both are nan
+        all_nan = np.all(
+            np.isnan(
+                [
+                    np_cum[
+                        o_2[0] : o_2[0] + lreg_shape[0], o_2[1] : o_2[1] + lreg_shape[1]
+                    ],
+                    np_viewshed,
+                ]
+            ),
+            axis=0,
+        )
+
+        np_cum[
+            o_2[0] : o_2[0] + lreg_shape[0], o_2[1] : o_2[1] + lreg_shape[1]
+        ] = np.nansum(
+            [
+                np_cum[
+                    o_2[0] : o_2[0] + lreg_shape[0], o_2[1] : o_2[1] + lreg_shape[1]
+                ],
+                np_viewshed,
+            ],
+            axis=0,
+        )
+
+        np_cum[o_2[0] : o_2[0] + lreg_shape[0], o_2[1] : o_2[1] + lreg_shape[1]][
+            all_nan
+        ] = np.nan
+
+        counter += 1
+
+    return np_cum
+
+
+def binary(
+    lreg_shape, t_loc, np_viewshed, reg, exp_range, r_dsm, dsm_type, v_elevation, b_1
```

Comment:
This is a very good suggestion, and I should have thought about it myself earlier. Unfortunately, now, this requires more significant code changes. Since this affects mainly code quality (and less performance or results) I would be glad if we could merge the current state and add this (maybe together with more informative variable names in the numpy sections) as a todo for later (after the PhD defense).

Using keyword arguments (and maybe functools.partial) could simplify the code quite a bit, I guess. Note to self: Here are some possible approaches:
- kwargs with starmap: https://stackoverflow.com/questions/45718523/pass-kwargs-to-starmap-while-using-pool-in-python
- solution with functools.partial: https://github.com/ipython/ipyparallel/issues/147

The functools.partial approach seems to be the most appropriate to me (with regards to improvements in code quality). But we would have to try out one or the other...